### PR TITLE
Convert BigNumber to  Strings and dates to ISO dates in ISeedConfig

### PR DESCRIPTION
**Pull request by Peartes**

### CODE REVIEW

File: src/newSeed/stage4.ts [hunk 1]
```diff
@@ -2,48 +2,57 @@ import { BaseStage } from "newSeed/baseStage";
 import Litepicker from "litepicker";
 
 export class Stage4 extends BaseStage {
-  protected startDate: HTMLElement | HTMLInputElement;
-  protected endDate: HTMLElement | HTMLInputElement;
+  protected startDateRef: HTMLElement | HTMLInputElement;
+  protected endDateRef: HTMLElement | HTMLInputElement;
+  protected startDate: { date: string, time: string };
```
[the remark this answers is not in the document]
Comment (Peartes) [reply src/newSeed/stage4.ts:7]: What do you mean ?

File: src/newSeed/stage4.html [hunk 2]
```diff
@@ -58,17 +80,22 @@
         <div>
           <label for="startDate">Start Date</label>
           <div class="vestingDates">
-            <input id="startDate" value.bind="seedConfig.seedDetails.startDate.date"
-              placeholder="DD-MM-YYYY" class="vestingDateInput" disabled>
-            <div><i class="far fa-calendar-alt" ref="startDate"></i></div>
+            <input 
+              id="startDate"
+              placeholder="DD-MM-YYYY"
```
Comment (dkent600) [inline src/newSeed/stage4.html:85]: is this the same format that litepicker uses?

File: src/newSeed/stage4.html [hunk 3]
```diff
@@ -81,17 +108,21 @@
         <div>
           <label for="endDate">End Date</label>
           <div class="vestingDates">
-            <input id="endDate" value.bind="seedConfig.seedDetails.endDate.date"
-            class="vestingDateInput" placeholder="DD-MM-YYY" disabled>
-            <div><i class="far fa-calendar-alt" ref="endDate"></i></div>
+            <input 
+              id="endDate"
+              class="vestingDateInput" 
+              placeholder="DD-MM-YYY"
```
Comment (dkent600) [inline src/newSeed/stage4.html:114]: needs four Y's, and be sure it matches the litepicker format

File: src/newSeed/stage4.ts [hunk 4]
```diff
@@ -1,49 +1,61 @@
+import { DateService } from "./../services/DateService";
 import { BaseStage } from "newSeed/baseStage";
 import Litepicker from "litepicker";
 
 export class Stage4 extends BaseStage {
-  protected startDate: HTMLElement | HTMLInputElement;
-  protected endDate: HTMLElement | HTMLInputElement;
+  protected startDateRef: HTMLElement | HTMLInputElement;
```
Comment (dkent600) [inline src/newSeed/stage4.ts:6]: remove `protected`s

File: src/newSeed/stage4.ts [hunk 5]
```diff
@@ -1,49 +1,61 @@
...
       message = "Please enter a non-zero value for the SEED Target";
-    } else if (!this.seedConfig.seedDetails.seedMax || this.seedConfig.seedDetails.seedMax.lte(0)) {
+    } else if (!this.seedConfig.seedDetails.seedMax || this.seedConfig.seedDetails.seedMax === "0") {
       message = "Please enter a non-zero number for the SEED Max";
     } else if (!this.seedConfig.seedDetails.vestingDays || this.seedConfig.seedDetails.vestingDays <= 0) {
       message = "Please enter a non-zero value for  \"Tokens vested for\" ";
     } else if (!this.seedConfig.seedDetails.vestingCliff || this.seedConfig.seedDetails.vestingCliff <= 0) {
       message = "Please enter a non-zero value for \"with a cliff of\" ";
-    } else if (!this.seedConfig.seedDetails.startDate) {
+    } else if (!this.startDate) {
```
Comment (dkent600) [inline src/newSeed/stage4.ts:52]: If it is possible to enter a malformed date, we should validate for it here.

File: src/newSeed/stage4.ts [hunk 6]
```diff
@@ -54,6 +66,15 @@ export class Stage4 extends BaseStage {
...
     } else {
+      // Set the ISO time
+      // Get the start and end time
+      const startTimes = this.startTime.split(":");
+      const endTimes = this.endTime.split(":");
+      this.startDate.setHours(Number.parseInt(startTimes[0]), Number.parseInt(startTimes[1]));
+      this.seedConfig.seedDetails.startDate = this.dateService.toISOString(this.startDate);
+      this.endDate.setHours(Number.parseInt(endTimes[0]), Number.parseInt(endTimes[1]));
+      this.seedConfig.seedDetails.endDate = this.dateService.toISOString(this.endDate);
+      console.log(this.seedConfig.seedDetails.endDate);
```
Comment (dkent600) [inline src/newSeed/stage4.ts:77]: remove the console.log

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: src/newSeed/stage4.ts (hunk 4 above)
Comment (dkent600) [reply src/newSeed/stage4.ts:6]: sorry I wasn't clear.  Remove "protected".  This is not a base class

File: src/newSeed/stage4.ts (hunk 5 above)
Comment (dkent600) [reply src/newSeed/stage4.ts:52]: Regarding time validation: Can the user enter a negative number, and would it be accepted?

Comment (Peartes) [reply src/newSeed/stage4.ts:52]: No the user cannot enter a negative number

**Review by dkent600 — APPROVED** [review]

**Pull request merged**